Clean up tsconfig references in Kibana

delanni opened the pull request:
## Summary
This PR removes path-based project references across typescript projects, and brings in a cleaner referencing, through package/plugin IDs. The ultimate goal was to have all ID references in our `kbn_references` sections.
 
This is a requirement for Moon recognising projects, and it's a cleaner and easier-to-parse way of tracking inter-package dependencies. 

## To reviewers
Feel free to review files changed under your ownership, here's some help to understand what changed. 

Changes generally should fall into a few categories only:
 - introduced `package.json`/`kibana.jsonc` for Kibana to appreciate a folder as a project and have it linked to `node_modules`
 - updated import references, whereas path-based references required file-relative imports, when packages are known by ID, they can be referred to by their package-root-relative imports.
 - updated `kbn_references` to existing/new package IDs
 - updated tooling code location, so they belong in packages

## Context
We've had several cases in our packages'/plugins' typescript configs, where one config references another config directly, like such:
```json
...
  "kbn_references": [
        ...
        { "path": "../setup_node_env/tsconfig.json" },
        ...
  ]
```
This instructs our type-checker abstraction, to generate the ad-hoc tsconfig for the project with a reference to the Typescript project located at `../setup_node_env/tsconfig.json`, setting up a dependency status.

In Moon, we'd like these dependencies to be also set up through the moon project structure, so we'd need a moon project to reference here. The reason why these odd `kbn_references` are made through paths, and not package IDs is because they're referring to code that has a `tsconfig.json`, and is a separate module of its' own, but it's not a package within Kibana, so it is not linked, and cannot be referenced by ID.

This PR aims to clear this up, by:
 - removing references that point to non-existent projects
 - removing `tsconfig.json`s where they're not necessary to delimit a separate project
 - introducing packages where necessary, so projects are identified, installed and can be referenced within Kibana
 - moving code to other folders, to collect them under named packages
 - updates references to newly named packages and import paths 
 
 
 Closes: https://github.com/elastic/kibana/issues/240141

<!--ONMERGE {"backportTargets":["8.19","9.1","9.2"]} ONMERGE-->

delanni committed:
remove references to non-existent typechecks

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/352498)
* Commit: b676ba46276b3eabab6a13eb4d0024c11e0abbfe

### Failed CI Steps
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/352498#019a00d8-d95d-497b-98ca-5f407c8ad62f)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/352498#019a010b-7894-49df-a530-07f5910862c5)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352498","number":352498,"commit":"b676ba46276b3eabab6a13eb4d0024c11e0abbfe"}],"number":352498}
buildkite-pr-comment-->

delanni committed:
Merge branch 'main' into clean-tsconfigs

delanni committed:
undo import merge

delanni committed:
move CLIs together

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_packages --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/generate codeowners'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

delanni committed:
setup security-solutions test utils with packages, move kbn_pm to dev/, erase unnecessary jest configs

delanni committed:
kbn_pm scripts should still use direct ref to setup_node_env

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/generate codeowners'

delanni committed:
setup-node-env is shared

delanni committed:
readd references to parent projects

delanni committed:
temporarily disable ts project linting

delanni committed:
fix imports in scripts/* to updated tool paths

delanni committed:
update instructions

delanni committed:
try to keep @kbn/setup-node-env in

delanni committed:
Merge branch 'main' into clean-tsconfigs

kibanamachine committed:
Changes from node scripts/generate codeowners

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

delanni committed:
when @kbn/setup-node-env is installed to node_modules, the root package.json will be at a different relative path

delanni committed:
use install-based imports to setup-node-env

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/356716)
* Commit: 87f092ee27c2bb58ec063431843bcbfc78360466

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/356716#019a3547-e4c4-4c7f-bc85-072c6659a28a)

### History
* :broken_heart: [Build #356255](https://buildkite.com/elastic/kibana-pull-request/builds/356255) failed 42d99b5ddaa09c2e6d87a1bc7ba20ee2ab2e8dca
* :broken_heart: [Build #356192](https://buildkite.com/elastic/kibana-pull-request/builds/356192) failed 492b2773d659671ccc87f9647fddbe7f360568c7
* :broken_heart: [Build #356166](https://buildkite.com/elastic/kibana-pull-request/builds/356166) failed e1c18860b2a83cee7742d868a98f73a87b32eda3
* :broken_heart: [Build #355855](https://buildkite.com/elastic/kibana-pull-request/builds/355855) failed af80e8bf07bf7632daaba56cdad0c732fee689b4
* :broken_heart: [Build #355473](https://buildkite.com/elastic/kibana-pull-request/builds/355473) failed af80e8bf07bf7632daaba56cdad0c732fee689b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356716","number":356716,"commit":"87f092ee27c2bb58ec063431843bcbfc78360466"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356255","number":356255,"commit":"42d99b5ddaa09c2e6d87a1bc7ba20ee2ab2e8dca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356192","number":356192,"commit":"492b2773d659671ccc87f9647fddbe7f360568c7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356166","number":356166,"commit":"e1c18860b2a83cee7742d868a98f73a87b32eda3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355855","number":355855,"commit":"af80e8bf07bf7632daaba56cdad0c732fee689b4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355473","number":355473,"commit":"af80e8bf07bf7632daaba56cdad0c732fee689b4"}],"number":356716}
buildkite-pr-comment-->

delanni committed:
weird hack, so dynamic imports won't trigger downstream import typechecks

delanni committed:
add dep to @kbn/setup-node-env, instead of the dynamic req hack

delanni committed:
Merge branch 'main' into clean-tsconfigs

delanni committed:
Merge branch 'main' into clean-tsconfigs

delanni committed:
remove last path references in tsconfigs, add @kbn/security_solution_api_integration as ts refs

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357279)
* Commit: 0c0a56e2912ea487f447dc28f211dbecbcad278d

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/357279#019a3b0e-13c9-456c-b19d-a27e7aa46f8c)

### History
* :broken_heart: [Build #356716](https://buildkite.com/elastic/kibana-pull-request/builds/356716) failed 87f092ee27c2bb58ec063431843bcbfc78360466
* :broken_heart: [Build #356255](https://buildkite.com/elastic/kibana-pull-request/builds/356255) failed 42d99b5ddaa09c2e6d87a1bc7ba20ee2ab2e8dca
* :broken_heart: [Build #356192](https://buildkite.com/elastic/kibana-pull-request/builds/356192) failed 492b2773d659671ccc87f9647fddbe7f360568c7
* :broken_heart: [Build #356166](https://buildkite.com/elastic/kibana-pull-request/builds/356166) failed e1c18860b2a83cee7742d868a98f73a87b32eda3
* :broken_heart: [Build #355855](https://buildkite.com/elastic/kibana-pull-request/builds/355855) failed af80e8bf07bf7632daaba56cdad0c732fee689b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357279","number":357279,"commit":"0c0a56e2912ea487f447dc28f211dbecbcad278d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356716","number":356716,"commit":"87f092ee27c2bb58ec063431843bcbfc78360466"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356255","number":356255,"commit":"42d99b5ddaa09c2e6d87a1bc7ba20ee2ab2e8dca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356192","number":356192,"commit":"492b2773d659671ccc87f9647fddbe7f360568c7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356166","number":356166,"commit":"e1c18860b2a83cee7742d868a98f73a87b32eda3"}],"number":357279}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357308)
* Commit: fd46d0cb2a26b0e17b64ae36e656a89aa3a420fa
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-239690)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### ESLint disabled in files
| id | [before](https://github.com/elastic/kibana/commit/21e34eabc452db0269eb48a0dcfc471f8d2e96a5) | [after](https://github.com/elastic/kibana/commit/fd46d0cb2a26b0e17b64ae36e656a89aa3a420fa) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 7 | +7 |
| `@kbn/test-suites-xpack-security` | 11 | 3 | -8 |
| `@kbn/test-suites-xpack-security-endpoint` | - | 1 | +1 |
| `inference` | 4 | 2 | -2 |
| total |  |  | -2 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/21e34eabc452db0269eb48a0dcfc471f8d2e96a5) | [after](https://github.com/elastic/kibana/commit/fd46d0cb2a26b0e17b64ae36e656a89aa3a420fa) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 22 | +22 |
| `@kbn/setup-node-env` | - | 3 | +3 |
| `@kbn/test-suites-xpack-security` | 96 | 74 | -22 |
| total |  |  | +3 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/21e34eabc452db0269eb48a0dcfc471f8d2e96a5) | [after](https://github.com/elastic/kibana/commit/fd46d0cb2a26b0e17b64ae36e656a89aa3a420fa) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 27 | +27 |
| `@kbn/test-suites-xpack-security` | 29 | 2 | -27 |
| total |  |  | -0 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/21e34eabc452db0269eb48a0dcfc471f8d2e96a5) | [after](https://github.com/elastic/kibana/commit/fd46d0cb2a26b0e17b64ae36e656a89aa3a420fa) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 29 | +29 |
| `@kbn/setup-node-env` | - | 3 | +3 |
| `@kbn/test-suites-xpack-security` | 107 | 77 | -30 |
| `@kbn/test-suites-xpack-security-endpoint` | - | 1 | +1 |
| `inference` | 6 | 4 | -2 |
| total |  |  | +1 |


</details>


### History
* :broken_heart: [Build #357279](https://buildkite.com/elastic/kibana-pull-request/builds/357279) failed 0c0a56e2912ea487f447dc28f211dbecbcad278d
* :broken_heart: [Build #356716](https://buildkite.com/elastic/kibana-pull-request/builds/356716) failed 87f092ee27c2bb58ec063431843bcbfc78360466
* :broken_heart: [Build #356255](https://buildkite.com/elastic/kibana-pull-request/builds/356255) failed 42d99b5ddaa09c2e6d87a1bc7ba20ee2ab2e8dca
* :broken_heart: [Build #356192](https://buildkite.com/elastic/kibana-pull-request/builds/356192) failed 492b2773d659671ccc87f9647fddbe7f360568c7
* :broken_heart: [Build #356166](https://buildkite.com/elastic/kibana-pull-request/builds/356166) failed e1c18860b2a83cee7742d868a98f73a87b32eda3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357308","number":357308,"commit":"fd46d0cb2a26b0e17b64ae36e656a89aa3a420fa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357279","number":357279,"commit":"0c0a56e2912ea487f447dc28f211dbecbcad278d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356716","number":356716,"commit":"87f092ee27c2bb58ec063431843bcbfc78360466"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356255","number":356255,"commit":"42d99b5ddaa09c2e6d87a1bc7ba20ee2ab2e8dca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356192","number":356192,"commit":"492b2773d659671ccc87f9647fddbe7f360568c7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356166","number":356166,"commit":"e1c18860b2a83cee7742d868a98f73a87b32eda3"}],"number":357308}
buildkite-pr-comment-->

delanni committed:
introduce a package for detections_response tools

kibanamachine committed:
Changes from node scripts/capture_oas_snapshot   --include-path /api/status   --include-path /api/alerting/rule/   --include-path /api/alerting/rules   --include-path /api/actions   --include-path /api/security/role   --include-path /api/spaces   --include-path /api/streams   --include-path /api/fleet   --include-path /api/saved_objects/_import   --include-path /api/saved_objects/_export   --include-path /api/maintenance_window   --include-path /api/agent_builder --update

kibanamachine committed:
Changes from node scripts/lint_packages --fix

kibanamachine committed:
Changes from node scripts/generate codeowners

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

mistic committed:
fix(NA): discover ts references when child ts projects exist on parent package and use relative paths

fix(NA): incorrect references on @kbn/lint-ts-projects for child ts projects inside a parent package

refact(NA): re-arrange and fix code

fix(NA: bugs in the implementation)

fix(NA): perf bugs

fix(NA): tsconfig

chore(NA): improved version

chore(NA): improved version

mistic committed:
chore(NA): merge and solve conflicts with main

mistic committed:
Merge branch 'main' into clean-tsconfigs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358787)
* Commit: 2fd3e5395deb960367a80828b4a54f17040794c3
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-239690)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/faf0c2852f9be73aa72572cfc92c635a9650a706) | [after](https://github.com/elastic/kibana/commit/2fd3e5395deb960367a80828b4a54f17040794c3) | diff |
| --- | --- | --- | --- |
| `@kbn/security-solution-detections-response-service` | - | 83 | +83 |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/faf0c2852f9be73aa72572cfc92c635a9650a706) | [after](https://github.com/elastic/kibana/commit/2fd3e5395deb960367a80828b4a54f17040794c3) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 1 | +1 |
| `alerting` | 61 | 62 | +1 |
| `securitySolution` | 39 | 42 | +3 |
| total |  |  | +5 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/faf0c2852f9be73aa72572cfc92c635a9650a706) | [after](https://github.com/elastic/kibana/commit/2fd3e5395deb960367a80828b4a54f17040794c3) | diff |
| --- | --- | --- | --- |
| `@kbn/security-solution-detections-response-service` | - | 119 | +119 |

#### ESLint disabled in files
| id | [before](https://github.com/elastic/kibana/commit/faf0c2852f9be73aa72572cfc92c635a9650a706) | [after](https://github.com/elastic/kibana/commit/2fd3e5395deb960367a80828b4a54f17040794c3) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 7 | +7 |
| `@kbn/test-suites-xpack-security` | 11 | 3 | -8 |
| `@kbn/test-suites-xpack-security-endpoint` | - | 1 | +1 |
| `inference` | 4 | 2 | -2 |
| total |  |  | -2 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/faf0c2852f9be73aa72572cfc92c635a9650a706) | [after](https://github.com/elastic/kibana/commit/2fd3e5395deb960367a80828b4a54f17040794c3) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 22 | +22 |
| `@kbn/setup-node-env` | - | 3 | +3 |
| `@kbn/test-suites-xpack-security` | 97 | 75 | -22 |
| total |  |  | +3 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/faf0c2852f9be73aa72572cfc92c635a9650a706) | [after](https://github.com/elastic/kibana/commit/2fd3e5395deb960367a80828b4a54f17040794c3) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 27 | +27 |
| `@kbn/test-suites-xpack-security` | 29 | 2 | -27 |
| total |  |  | -0 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/faf0c2852f9be73aa72572cfc92c635a9650a706) | [after](https://github.com/elastic/kibana/commit/2fd3e5395deb960367a80828b4a54f17040794c3) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 29 | +29 |
| `@kbn/setup-node-env` | - | 3 | +3 |
| `@kbn/test-suites-xpack-security` | 108 | 78 | -30 |
| `@kbn/test-suites-xpack-security-endpoint` | - | 1 | +1 |
| `inference` | 6 | 4 | -2 |
| total |  |  | +1 |


</details>


<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358787","number":358787,"commit":"2fd3e5395deb960367a80828b4a54f17040794c3"}],"number":358787}
buildkite-pr-comment-->

delanni committed:
unskip lint_ts_projects quick check

delanni committed:
add fleet-plugin-cypress, update dep path in x-pack/platform/plugins/shared/fleet/cypress/tasks/agents.ts

delanni committed:
introduce packages for osquery/cypress and fleet/cypress

delanni committed:
Merge branch 'main' into clean-tsconfigs

mistic committed:
fix(NA): include current missing package on package.json

mistic committed:
revert(NA): changes on packages/kbn-lint-ts-projects-cli/rules/reference_used_pkgs.ts and removed self reference on cypress pkg

kibanamachine committed:
Changes from node scripts/lint_packages --fix

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

kibanamachine committed:
Changes from node scripts/generate codeowners

delanni committed:
Merge branch 'main' into clean-tsconfigs

mistic committed:
fix(NA): cypress fleet configs

mistic committed:
Merge remote-tracking branch 'delanni/clean-tsconfigs' into delanni-clean-tsconfigs-fixed

mistic committed:
fix(NA): cypress config paths for fleet plugin

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/359386)
* Commit: dc3e0b92be15dc5066565b60f2a688bfe857801c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-239690)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/2a235bf388e30604ba19ff57bc4234b8f4c1efed) | [after](https://github.com/elastic/kibana/commit/dc3e0b92be15dc5066565b60f2a688bfe857801c) | diff |
| --- | --- | --- | --- |
| `@kbn/security-solution-detections-response-service` | - | 83 | +83 |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/2a235bf388e30604ba19ff57bc4234b8f4c1efed) | [after](https://github.com/elastic/kibana/commit/dc3e0b92be15dc5066565b60f2a688bfe857801c) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 1 | +1 |
| `alerting` | 61 | 62 | +1 |
| `securitySolution` | 39 | 42 | +3 |
| total |  |  | +5 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/2a235bf388e30604ba19ff57bc4234b8f4c1efed) | [after](https://github.com/elastic/kibana/commit/dc3e0b92be15dc5066565b60f2a688bfe857801c) | diff |
| --- | --- | --- | --- |
| `@kbn/security-solution-detections-response-service` | - | 119 | +119 |

#### ESLint disabled in files
| id | [before](https://github.com/elastic/kibana/commit/2a235bf388e30604ba19ff57bc4234b8f4c1efed) | [after](https://github.com/elastic/kibana/commit/dc3e0b92be15dc5066565b60f2a688bfe857801c) | diff |
| --- | --- | --- | --- |
| `@kbn/osquery-plugin-cypress` | - | 1 | +1 |
| `@kbn/security_solution_api_integration` | - | 7 | +7 |
| `@kbn/test-suites-xpack-security` | 11 | 3 | -8 |
| `@kbn/test-suites-xpack-security-endpoint` | - | 1 | +1 |
| `inference` | 4 | 2 | -2 |
| `osquery` | 1 | 0 | -1 |
| total |  |  | -2 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/2a235bf388e30604ba19ff57bc4234b8f4c1efed) | [after](https://github.com/elastic/kibana/commit/dc3e0b92be15dc5066565b60f2a688bfe857801c) | diff |
| --- | --- | --- | --- |
| `@kbn/fleet-plugin-cypress` | - | 4 | +4 |
| `@kbn/osquery-plugin-cypress` | - | 4 | +4 |
| `@kbn/security_solution_api_integration` | - | 22 | +22 |
| `@kbn/setup-node-env` | - | 3 | +3 |
| `@kbn/test-suites-xpack-security` | 97 | 75 | -22 |
| `fleet` | 45 | 41 | -4 |
| `osquery` | 93 | 89 | -4 |
| total |  |  | +3 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/2a235bf388e30604ba19ff57bc4234b8f4c1efed) | [after](https://github.com/elastic/kibana/commit/dc3e0b92be15dc5066565b60f2a688bfe857801c) | diff |
| --- | --- | --- | --- |
| `@kbn/security_solution_api_integration` | - | 27 | +27 |
| `@kbn/test-suites-xpack-security` | 29 | 2 | -27 |
| total |  |  | -0 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/2a235bf388e30604ba19ff57bc4234b8f4c1efed) | [after](https://github.com/elastic/kibana/commit/dc3e0b92be15dc5066565b60f2a688bfe857801c) | diff |
| --- | --- | --- | --- |
| `@kbn/fleet-plugin-cypress` | - | 4 | +4 |
| `@kbn/osquery-plugin-cypress` | - | 5 | +5 |
| `@kbn/security_solution_api_integration` | - | 29 | +29 |
| `@kbn/setup-node-env` | - | 3 | +3 |
| `@kbn/test-suites-xpack-security` | 108 | 78 | -30 |
| `@kbn/test-suites-xpack-security-endpoint` | - | 1 | +1 |
| `fleet` | 57 | 53 | -4 |
| `inference` | 6 | 4 | -2 |
| `osquery` | 94 | 89 | -5 |
| total |  |  | +1 |


</details>


### History
* :broken_heart: [Build #359309](https://buildkite.com/elastic/kibana-pull-request/builds/359309) failed 016c07cf67cfa29e9247f7a2b4bb3819747f5f28
* :broken_heart: [Build #359226](https://buildkite.com/elastic/kibana-pull-request/builds/359226) failed 4bfe135bd1a6017d3ae2c6d5bf107401bcea87d4
* :broken_heart: [Build #359210](https://buildkite.com/elastic/kibana-pull-request/builds/359210) failed bedac982c7db03c5784716aa8551799840d083a3
* :green_heart: [Build #358787](https://buildkite.com/elastic/kibana-pull-request/builds/358787) succeeded 2fd3e5395deb960367a80828b4a54f17040794c3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359386","number":359386,"commit":"dc3e0b92be15dc5066565b60f2a688bfe857801c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359309","number":359309,"commit":"016c07cf67cfa29e9247f7a2b4bb3819747f5f28"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359226","number":359226,"commit":"4bfe135bd1a6017d3ae2c6d5bf107401bcea87d4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359210","number":359210,"commit":"bedac982c7db03c5784716aa8551799840d083a3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358787","number":358787,"commit":"2fd3e5395deb960367a80828b4a54f17040794c3"}],"number":359386}
buildkite-pr-comment-->

delanni committed:
correct cypress owners

nreese reviewed (approved):
kibana-presentation changes LGTM
code review only

CoenWarmer reviewed (approved):
Profiling changes LGTM

delanni committed:
Merge branch 'main' into clean-tsconfigs

pmuellr reviewed (approved):
ResponseOps changes LGTM

PhilippeOberti reviewed (approved):
Code review only, LGTM for the @elastic/security-threat-hunting team

delanni committed:
Merge branch 'main' into clean-tsconfigs

gergoabraham reviewed (approved):
@elastic/security-defend-workflows changes LGTM, thanks! 🚀

maximpn reviewed (approved):
Rule Management Area changes LGTM

rylnd commented on x-pack/solutions/security/test/security_solution_api_integration/kibana.jsonc:
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "test-helper",
+  "id": "@kbn/security_solution_api_integration",
```
This package is mainly FTR tests, with the sub-package above containing the test helpers. Can we rename this and/or add a new package type to reflect this fact?

```suggestion
  "type": "functional-tests",
  "id": "@kbn/security-solution-api-ftr-tests",
```

rylnd replied:
Or: I see other packages follow the `@kbn/test-suites-` prefix; so maybe `@kbn/test-suites-security-solution-apis` if that's your preference?

rylnd commented on x-pack/solutions/security/test/security_solution_api_integration/config/services/detections_response/kibana.jsonc:
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "test-helper",
+  "id": "@kbn/security-solution-detections-response-service",
```
Nit: as this is a subpackage of the below, I think this could be named more accurately:

```suggestion
  "type": "test-helper",
  "id": "@kbn/detections-response-ftr-services",
```

rylnd reviewed (approved):
Detection Engine changes LGTM; just had a few nits around naming/categorization of extracted packages.

delanni committed:
Merge branch 'main' into clean-tsconfigs

delanni committed:
rename packages

delanni committed:
Merge branch 'main' into clean-tsconfigs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/360587)
* Commit: 09971dba581856498b5a8479e74614afbe373a8b

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/360587#019a7883-e26b-4d2f-b887-9111a7e90099)
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/360587#019a7883-e26c-4224-b400-596612cc21ec)

### History
* :broken_heart: [Build #360068](https://buildkite.com/elastic/kibana-pull-request/builds/360068) failed 1e01446c5197779e46c018021a8412481a5ef87f
* :broken_heart: [Build #359906](https://buildkite.com/elastic/kibana-pull-request/builds/359906) failed 28b56abe2013ac44e02bfc6c8a7e4662a2c0a935
* :green_heart: [Build #359386](https://buildkite.com/elastic/kibana-pull-request/builds/359386) succeeded dc3e0b92be15dc5066565b60f2a688bfe857801c
* :broken_heart: [Build #359309](https://buildkite.com/elastic/kibana-pull-request/builds/359309) failed 016c07cf67cfa29e9247f7a2b4bb3819747f5f28
* :broken_heart: [Build #359226](https://buildkite.com/elastic/kibana-pull-request/builds/359226) failed 4bfe135bd1a6017d3ae2c6d5bf107401bcea87d4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360587","number":360587,"commit":"09971dba581856498b5a8479e74614afbe373a8b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360068","number":360068,"commit":"1e01446c5197779e46c018021a8412481a5ef87f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359906","number":359906,"commit":"28b56abe2013ac44e02bfc6c8a7e4662a2c0a935"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359386","number":359386,"commit":"dc3e0b92be15dc5066565b60f2a688bfe857801c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359309","number":359309,"commit":"016c07cf67cfa29e9247f7a2b4bb3819747f5f28"}],"number":360587}
buildkite-pr-comment-->

kibanamachine committed:
Changes from node scripts/capture_oas_snapshot   --include-path /api/status   --include-path /api/alerting/rule/   --include-path /api/alerting/rules   --include-path /api/actions   --include-path /api/security/role   --include-path /api/spaces   --include-path /api/streams   --include-path /api/fleet   --include-path /api/saved_objects/_import   --include-path /api/saved_objects/_export   --include-path /api/maintenance_window   --include-path /api/agent_builder --update

stratoula reviewed (approved):
Only one comma added

kibanamachine committed:
TO FIX: Run node 'scripts/yarn_deduplicate && yarn kbn bootstrap' locally, or add an exception to src/dev/yarn_deduplicate/index.ts and then commit the changes and push to your branch

kibanamachine committed:
Changes from node scripts/generate codeowners

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

jgowdyelastic reviewed (approved):
ML changes LGTM

delanni committed:
Merge branch 'main' into clean-tsconfigs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/361112)
* Commit: 980570d570e267b1987698ccb270649ac2eaebd0
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-239690)

### Failed CI Steps
* [Serverless AI4DSOC - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/361112#019a7e1f-e3ca-4cea-9baa-ea1bde9cb563)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/361112#019a7e1f-e3ca-4cea-9baa-ea1bde9cb563) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/361112/jobs/019a7e1f-e3ca-4cea-9baa-ea1bde9cb563/artifacts/019a7e55-d04a-4dd4-bd94-7944cab7bcc5) Serverless AI4DSOC - Security Solution Cypress Tests / AI4dSoC Navigation renders pages within links correctly should show the correct page for visible links when navigating should show the correct page for visible links when navigating

### History
* :yellow_heart: [Build #360648](https://buildkite.com/elastic/kibana-pull-request/builds/360648) was flaky 3fc418bb38359d74f74556bbc64c2fb8a68bc426
* :broken_heart: [Build #360587](https://buildkite.com/elastic/kibana-pull-request/builds/360587) failed 09971dba581856498b5a8479e74614afbe373a8b
* :broken_heart: [Build #360068](https://buildkite.com/elastic/kibana-pull-request/builds/360068) failed 1e01446c5197779e46c018021a8412481a5ef87f
* :broken_heart: [Build #359906](https://buildkite.com/elastic/kibana-pull-request/builds/359906) failed 28b56abe2013ac44e02bfc6c8a7e4662a2c0a935
* :green_heart: [Build #359386](https://buildkite.com/elastic/kibana-pull-request/builds/359386) succeeded dc3e0b92be15dc5066565b60f2a688bfe857801c
* :broken_heart: [Build #359309](https://buildkite.com/elastic/kibana-pull-request/builds/359309) failed 016c07cf67cfa29e9247f7a2b4bb3819747f5f28

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361112","number":361112,"commit":"980570d570e267b1987698ccb270649ac2eaebd0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360648","number":360648,"commit":"3fc418bb38359d74f74556bbc64c2fb8a68bc426"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360587","number":360587,"commit":"09971dba581856498b5a8479e74614afbe373a8b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360068","number":360068,"commit":"1e01446c5197779e46c018021a8412481a5ef87f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359906","number":359906,"commit":"28b56abe2013ac44e02bfc6c8a7e4662a2c0a935"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359386","number":359386,"commit":"dc3e0b92be15dc5066565b60f2a688bfe857801c"}],"number":361112}
buildkite-pr-comment-->

delanni committed:
Merge branch 'main' into clean-tsconfigs

delanni committed:
Merge branch 'main' into clean-tsconfigs

delanni committed:
add missing owners to new packages

delanni committed:
Merge branch 'main' into clean-tsconfigs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/361759)
* Commit: f46acc5e0e396a6f3d7a766857eeb952667e0b6e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-239690)

### Failed CI Steps
* [Defend Workflows Cypress Tests on Serverless #3](https://buildkite.com/elastic/kibana-pull-request/builds/361759#019a9150-2a30-4f00-86bd-411199f7894f)

### History
* :green_heart: [Build #361368](https://buildkite.com/elastic/kibana-pull-request/builds/361368) succeeded eb7f0584c1ab519e9c88972e68fbb4b6f7323a71
* :broken_heart: [Build #361273](https://buildkite.com/elastic/kibana-pull-request/builds/361273) failed 3384a4672a5ebf2c28e353a4c96ce11e6f3e9032
* :broken_heart: [Build #361112](https://buildkite.com/elastic/kibana-pull-request/builds/361112) failed 980570d570e267b1987698ccb270649ac2eaebd0
* :yellow_heart: [Build #360648](https://buildkite.com/elastic/kibana-pull-request/builds/360648) was flaky 3fc418bb38359d74f74556bbc64c2fb8a68bc426
* :broken_heart: [Build #360587](https://buildkite.com/elastic/kibana-pull-request/builds/360587) failed 09971dba581856498b5a8479e74614afbe373a8b
* :broken_heart: [Build #360068](https://buildkite.com/elastic/kibana-pull-request/builds/360068) failed 1e01446c5197779e46c018021a8412481a5ef87f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361759","number":361759,"commit":"f46acc5e0e396a6f3d7a766857eeb952667e0b6e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361368","number":361368,"commit":"eb7f0584c1ab519e9c88972e68fbb4b6f7323a71"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361273","number":361273,"commit":"3384a4672a5ebf2c28e353a4c96ce11e6f3e9032"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361112","number":361112,"commit":"980570d570e267b1987698ccb270649ac2eaebd0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360648","number":360648,"commit":"3fc418bb38359d74f74556bbc64c2fb8a68bc426"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360587","number":360587,"commit":"09971dba581856498b5a8479e74614afbe373a8b"}],"number":361759}
buildkite-pr-comment-->

delanni committed:
Merge branch 'main' into clean-tsconfigs

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.19|Backport failed because of merge conflicts|
|❌|9.1|Backport failed because of merge conflicts|
|❌|9.2|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 239690
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 239690` locally
cc: @delanni

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 239690` locally
cc: @delanni

delanni commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.2|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/244348">](https://github.com/elastic/kibana/pull/244348)|
|✅|9.1|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/244369">](https://github.com/elastic/kibana/pull/244369)|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/244458">](https://github.com/elastic/kibana/pull/244458)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.2.0 -->
